### PR TITLE
HRINT-3161 Prepare scripts for changes in RavenDB 6.0 builds

### PR DESCRIPTION
**Pull request by hiddenshadow21**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -8,22 +8,28 @@ on:
         description: 'RavenDB Version'
         required: true
         type: string
+      triggering_build_number:
```
Comment (gregolsky) [inline .github/workflows/build.yml]: what do you need this for?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -33,17 +39,26 @@ jobs:
           echo "RAVEN_MAJOR_MINOR=$v" >> $GITHUB_ENV
           [[ ${{ env.IS_NIGHTLY }} == true ]] && DOCKER_REPO="$DOCKER_REPO-nightly"
           echo "DOCKER_REPO=$DOCKER_REPO" >> $GITHUB_ENV
+          buildType=$(echo ${{ inputs.version }} | cut -d- -f2)
```
Comment (gregolsky) [inline .github/workflows/build.yml:53]: Please also log all those parameters to the console - build type, major minor, docker repo and the build number

File: s3uploader.sh [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+# original usage
+# S3Uploader.exe [category] [buildVersion] [filename] [comment]
+
+# settings file
+#
+# UrlToPostTo = ConfigurationManager.AppSettings["UrlToPostTo"],
+# SecretPass = ConfigurationManager.AppSettings["SecretPass"],
+# BucketName = ConfigurationManager.AppSettings["BucketName"],
+# AwsAccessKey = ConfigurationManager.AppSettings["AwsAccessKey"],
+# AwsSecretKey = ConfigurationManager.AppSettings["AwsSecretKey"],
```
Comment (gregolsky) [inline s3uploader.sh]: Please remove these comments

File: .github/workflows/build.yml (hunk 1 above)
Comment (hiddenshadow21) [reply .github/workflows/build.yml]: It is needed for request to our API.

Comment (gregolsky) [reply .github/workflows/build.yml]: we talked

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -61,17 +76,138 @@ jobs:
...
+        uses: actions/upload-artifact@v3
+        with:
+          name: package-${{ matrix.ubuntuName }}-${{ matrix.arch }}
+          path: ravendb/scripts/linux/pkg/deb/dist/${{ env.UBUNTU_VERSION }}/ravendb*.deb
+
+      - name: Upload deb package to s3
+        run: |
+          set -e
+          set -x
+          #export DRY_RUN=1
```
Comment (gregolsky) [inline .github/workflows/build.yml]: can you allow passing dry_run as the top-level GA action parameter?

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -61,17 +76,138 @@ jobs:
...
+          ls -R $PACKAGES_DIST_DIR
+
+          if [ "${{ env.IS_NIGHTLY }}" == "true" ]; then
+
+            NIGHTLY_VERSION_DATE_PART=$(date "+%Y%M%d-%H%m" -u)
+            export BUILD_VERSION="$NIGHTLY_VERSION_DATE_PART-Nightly"
+          elif [ "${{ env.IS_STABLE }}" == "true" ]; then
+            export BUILD_VERSION="${{ inputs.triggering_build_number }}"
+          else
+            export BUILD_VERSION="${{ inputs.triggering_build_number }}-${{ env.TRIGGERING_BUILD_TYPE }}"
```
Comment (gregolsky) [inline .github/workflows/build.yml]: We talked that triggering_build_number is something different and you need something else.

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -61,17 +76,138 @@ jobs:
...
+          fi
+
+          export COMMENT="RavenDB_${RAVENDB_VERSION}_DEB"
+          export S3UPLOADER="./current/s3uploader.sh"
+
+          "$S3UPLOADER" --category "RavenDB for Ubuntu ${{ env.UBUNTU_VERSION }} ${{ matrix.arch }} DEB"   --filename "${PACKAGES_DIST_DIR}/${{ env.UBUNTU_VERSION }}/ravendb_${RAVENDB_VERSION}-0_ubuntu.${{ env.UBUNTU_VERSION }}_${{ env.DEB_ARCH }}.deb"
```
Comment (gregolsky) [inline .github/workflows/build.yml]: can you break lines on parameters?

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -61,17 +76,138 @@ jobs:
...
+      - name: Get ravendb majorminor version & set repo
+        run: |
+          v=$(echo ${{ inputs.version }} | cut -d. -f1,2)
+          echo "RAVEN_MAJOR_MINOR=$v" >> $GITHUB_ENV
+          [[ ${{ env.IS_NIGHTLY }} == true ]] && DOCKER_REPO="$DOCKER_REPO-nightly"
+          echo "DOCKER_REPO=$DOCKER_REPO" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v3
+        with:
+          repository: 'hiddenshadow21/ravendb'
```
Comment (gregolsky) [inline .github/workflows/build.yml]: can we make that a job parameter defaulting to ravendb/ravendb?

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -61,17 +76,138 @@ jobs:
...
+          ref: "v${{ env.RAVEN_MAJOR_MINOR }}"
+
+      - uses: actions/download-artifact@v3
+        with:
+          name: package-focal-${{ matrix.arch }}
+          path: downloads
+
+      - name: Move artifacts to correct folder
```
Comment (gregolsky) [inline .github/workflows/build.yml]: Doesn't docker build script take files from the `artifacts` dir by default?

File: .github/workflows/build.yml [hunk 9]
```diff
@@ -61,17 +76,138 @@ jobs:
...
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to docker
+        run: |
+          docker login -u ${{ env.DOCKER_USER }} -p ${{ secrets.DOCKER_HUB_PASSWORD }}
```
Comment (gregolsky) [inline .github/workflows/build.yml:199]: We can pass an API token for DOCKER_HUB_PASSWORD, right?

**Review by gregolsky — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 8 above)
Comment (hiddenshadow21) [reply .github/workflows/build.yml]: Only `tar` files, `.deb` by default are in `docker/ravendb-ubuntu/` dir.

File: .github/workflows/build.yml (hunk 9 above)
Comment (hiddenshadow21) [reply .github/workflows/build.yml:199]: Yes

File: .github/workflows/build.yml (hunk 2 above)
Comment (hiddenshadow21) [reply .github/workflows/build.yml:53]: Done.

File: s3uploader.sh (hunk 3 above)
Comment (hiddenshadow21) [reply s3uploader.sh]: Removed.

File: .github/workflows/build.yml (hunk 4 above)
Comment (hiddenshadow21) [reply .github/workflows/build.yml]: Added.

File: .github/workflows/build.yml (hunk 6 above)
Comment (hiddenshadow21) [reply .github/workflows/build.yml]: Done.

File: .github/workflows/build.yml (hunk 5 above)
Comment (hiddenshadow21) [reply .github/workflows/build.yml]: Changed.

File: .github/workflows/build.yml (hunk 7 above)
Comment (hiddenshadow21) [reply .github/workflows/build.yml]: Done.

**Pull request merged**